File: services/notifications/notifications-service/src/main/java/razepl/dev/social365/notifications/api/notifications/interfaces/NotificationsService.java
package razepl.dev.social365.notifications.api.notifications.interfaces;

import org.springframework.data.domain.Pageable;
import razepl.dev.social365.notifications.api.notifications.data.NotificationResponse;
import razepl.dev.social365.notifications.config.auth.User;
import razepl.dev.social365.notifications.utils.pagination.SocialPage;

public interface NotificationsService {

    SocialPage<NotificationResponse> getNotificationsForUser(User user, Pageable pageable);

    NotificationResponse readNotifications(User user);

}

File: services/frontend-client/src/app/core/services/api/notifications/notifications.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { NotificationsMappings } from "@enums/api/notifications/notifications-mappings.enum";
import { map, Observable, take } from "rxjs";
import { SocialPage } from "@core/utils/social-page";
import { PageablePagingState } from "@core/utils/pageable-paging-state";
import { Notification } from "@interfaces/notifications/notification.interface";

@Injectable({
    providedIn: 'root'
})
export class NotificationsService {

    constructor(private http: HttpClient) {
    }

    getNotificationsForUser(pagingState: PageablePagingState): Observable<SocialPage<Notification, PageablePagingState>> {
        return this.http.get<SocialPage<Notification, PageablePagingState>>(NotificationsMappings.NOTIFICATIONS_MAPPING, {
            params: {
                pageNumber: pagingState.pageNumber,
                pageSize: pagingState.pageSize
            }
        })
            .pipe(
                take(1),
                map(json => SocialPage.fromJson<Notification, PageablePagingState>(json, PageablePagingState.fromJson, (val) => val as Notification))
            );
    }

    readNotifications(): Observable<void> {
        return this.http.put<void>(NotificationsMappings.READ_NOTIFICATIONS_MAPPING, {}).pipe(take(1));
    }

}
